src/test/java/com/puppycrawl/tools/checkstyle/ant/CheckstyleAntTaskTest.java

```diff
@@ -767,27 +770,75 @@ public void testMaxWarnings() throws IOException {
                 .isTrue();
     }
 
-    private static List<String> readWholeFile(File outputFile) throws IOException {
-        return Files.readAllLines(outputFile.toPath(), StandardCharsets.UTF_8);
-    }
+    @Test
+    public final void testExecuteLogOutput() throws Exception {
+        final CheckstyleAntTaskLogStub antTask = new CheckstyleAntTaskLogStub();
+        final URL url = new File(getPath(CONFIG_FILE)).toURI().toURL();
+        antTask.setProject(new Project());
+        antTask.setConfig(url.toString());
+        antTask.setFile(new File(getPath(FLAWLESS_INPUT)));
 
-    private static class CheckstyleAntTaskLogStub extends CheckstyleAntTask {
+        antTask.execute();
 
-        private final List<String> loggedMessages = new ArrayList<>();
+        final Violation auditStartedMessage = new Violation(1,
+                Definitions.CHECKSTYLE_BUNDLE, "DefaultLogger.auditStarted",
+                null, null,
+                getClass(), null);
+        final Violation auditFinishedMessage = new Violation(1,
+                Definitions.CHECKSTYLE_BUNDLE, "DefaultLogger.auditFinished",
+                null, null,
+                getClass(), null);
 
-        @Override
-        public void log(String msg, int msgLevel) {
-            loggedMessages.add(msg);
+        final List<MessageLevelPair> expectedList = Arrays.asList(
+                new MessageLevelPair("checkstyle version .*", Project.MSG_VERBOSE),
+                new MessageLevelPair("Adding standalone file for audit", Project.MSG_VERBOSE),
+                new MessageLevelPair("To locate the files took \\d+ ms.", Project.MSG_VERBOSE),
+                new MessageLevelPair("Running Checkstyle  on 1 files", Project.MSG_INFO),
+                new MessageLevelPair("Using configuration file:.*", Project.MSG_VERBOSE),
+                new MessageLevelPair(auditStartedMessage.getViolation(), Project.MSG_DEBUG),
+                new MessageLevelPair(auditFinishedMessage.getViolation(), Project.MSG_DEBUG),
+                new MessageLevelPair("To process the files took \\d+ ms.", Project.MSG_VERBOSE),
+                new MessageLevelPair("Total execution took \\d+ ms.", Project.MSG_VERBOSE)
+        );
+
+        final List<MessageLevelPair> loggedMessages = antTask.getLoggedMessages();
```

Comment:
lets move this close to execute:
```


        final Violation auditStartedMessage = new Violation(1,
                Definitions.CHECKSTYLE_BUNDLE, "DefaultLogger.auditStarted",
                null, null,
                getClass(), null);
        final Violation auditFinishedMessage = new Violation(1,
                Definitions.CHECKSTYLE_BUNDLE, "DefaultLogger.auditFinished",
                null, null,
                getClass(), null);

        final List<MessageLevelPair> expectedList = Arrays.asList(
                new MessageLevelPair("checkstyle version .*", Project.MSG_VERBOSE),
                new MessageLevelPair("Adding standalone file for audit", Project.MSG_VERBOSE),
                new MessageLevelPair("To locate the files took \\d+ ms.", Project.MSG_VERBOSE),
                new MessageLevelPair("Running Checkstyle  on 1 files", Project.MSG_INFO),
                new MessageLevelPair("Using configuration file:.*", Project.MSG_VERBOSE),
                new MessageLevelPair(auditStartedMessage.getViolation(), Project.MSG_DEBUG),
                new MessageLevelPair(auditFinishedMessage.getViolation(), Project.MSG_DEBUG),
                new MessageLevelPair("To process the files took \\d+ ms.", Project.MSG_VERBOSE),
                new MessageLevelPair("Total execution took \\d+ ms.", Project.MSG_VERBOSE)
        );

        antTask.execute();
        final List<MessageLevelPair> loggedMessages = antTask.getLoggedMessages();
```

so it will be like setup Task, setup expected, execute, validate

Reply:
done